### PR TITLE
[bgp] Save bgp admin state

### PR DESCRIPTION
**Pull request by taoyl-ms**

### CODE REVIEW

File: dockers/docker-fpm-quagga/start.sh [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/usr/bin/env bash
 
 mkdir -p /etc/quagga
-sonic-cfggen -m /etc/sonic/minigraph.xml -t /usr/share/sonic/templates/bgpd.conf.j2 > /etc/quagga/bgpd.conf
+if [ -f /etc/sonic/bgp_admin.yml ]; then
+    sonic-cfggen -m /etc/sonic/minigraph.xml -y /etc/sonic/bgp_admin.yml -t /usr/share/sonic/templates/bgpd.conf.j2 > /etc/quagga/bgpd.conf
```
Comment (stcheng) [inline dockers/docker-fpm-quagga/start.sh]: a sample file of bgp_admin.yml file could be provided in sonic-cfggen tests folder.

File: dockers/docker-fpm-quagga/base_image_files/bgp_neighbor [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+fi
+
+ASN=`vtysh -c "show ip bgp su" | sed -n "s/.*AS number \([0-9]\+\).*/\1/p"`
+if [ -z "$ASN" ]; then
+    exit 255
+fi
+
+[ -f /etc/sonic/bgp_admin.yml ] || echo "bgp_admin_state:" > /etc/sonic/bgp_admin.yml
+
+if [ "$NEIGHBOR_IP" == "0.0.0.0" ] ; then
```
Comment (lguohan) [inline dockers/docker-fpm-quagga/base_image_files/bgp_neighbor]: add comment: "shutdown down all ipv4 neighbors when "neighbor_ip" = 0.0.0.0

File: dockers/docker-fpm-quagga/base_image_files/bgp_neighbor [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+if [ "$COMMAND" == "shutdown" ]; then
+    CMD_PREFIX=""
+elif [ "$COMMAND" == "startup" ]; then
+    CMD_PREFIX="no"
+else
+    usage
+fi
+
+ASN=`vtysh -c "show ip bgp su" | sed -n "s/.*AS number \([0-9]\+\).*/\1/p"`
```
Comment (lguohan) [inline dockers/docker-fpm-quagga/base_image_files/bgp_neighbor]: can we use the full names here. su -> summary

**Comment by lguohan** [comment]
why not change this file https://github.com/Azure/sonic-utilities/blob/master/config/main.py to add bgp neighbor shut/unshut functionality? why add another script to do this? I feel this utilities should be part of sonic-utilities.

**Comment by taoyl-ms** [comment]
I agree. Planned to add into sonic-utilities later.

**Comment by taoyl-ms** [comment]
Close it by now as we might need more discussion on the default behavior of bgp admin state.

**Comment by taoyl-ms** [comment]
Set default admin state to on and add build option to support default off.

**Pull request merged**